scipy/stats/tests/test_bootstrap.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import pytest
+from scipy.stats import bootstrap_ci
+from numpy.testing import assert_allclose
+from scipy import stats
+from .. import _bootstrap as bootstrap
+from scipy._lib._util import rng_integers
+
+
+def test_bootstrap_ci_iv():
+    message = "`data` must be a sequence of samples."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(1, np.mean)
+
+    message = "`data` must contain at least one sample."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(tuple(), np.mean)
+
+    message = "each sample in `data` must contain two or more observations..."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3], [1]), np.mean)
+
+    message = "`axis` must be an integer."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, axis=1.5)
+
+    message = "could not convert string to float"
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, confidence_level='ni')
+
+    message = "`n_resamples` must be a positive integer."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, n_resamples=-1000)
+
+    message = "`n_resamples` must be a positive integer."
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, n_resamples=1000.5)
+
+    message = "`method` must be in"
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, method='ekki')
+
+    message = "`method = 'BCa' is only available for one-sample statistics"
+
+    def statistic(x, y, axis):
+        mean1 = np.mean(x, axis)
+        mean2 = np.mean(y, axis)
+        return mean1 - mean2
+
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([.1, .2, .3], [.1, .2, .3]), statistic, method='BCa')
+
+    message = "'herring' cannot be used to seed a"
+    with pytest.raises(ValueError, match=message):
+        bootstrap_ci(([1, 2, 3],), np.mean, random_state='herring')
+
+
+@pytest.mark.parametrize("method", ['basic', 'percentile', 'BCa'])
+def test_bootstrap_ci_against_theory(method):
+    # based on https://www.statology.org/confidence-intervals-python/
+    data = stats.norm.rvs(loc=5, scale=2, size=5000)
+    alpha = 0.95
+    expected = stats.t.interval(alpha=alpha, df=len(data)-1,
+                                loc=np.mean(data), scale=stats.sem(data))
+    res = bootstrap_ci((data,), np.mean, n_resamples=5000,
+                       confidence_level=alpha, method=method,
+                       random_state=0)
+    assert_allclose(res, expected, 6e-4)
+
+
+tests_R = {"basic": (23.77, 79.12),
+           "percentile": (28.86, 84.21),
+           "BCa": (32.31, 91.43)}
+
+
+@pytest.mark.parametrize("method, expected", tests_R.items())
+def test_bootstrap_ci_against_R(method, expected):
```

Comment:
Do we need this one considering the other tests? Not much of an overhead but if we can limit the number of tests.

Reply:
Warren gave this example in his review, and I was very happy to see the agreement. The theory one above is nice, but these are the only two that compare against an outside reference, and this one is stronger, IMO. I think it's important to keep.

Reply:
ok sure no problem 👍